app/controllers/api/v1/visualization/expression_controller.rb

```diff
@@ -51,6 +53,28 @@ def render_violin
           render json: render_data, status: 200
         end
 
+        def render_heatmap
+          cluster = RequestUtils.get_cluster_group(params, @study)
```

Comment:
This method is in the `ClusterVizService` now.
```suggestion
          cluster = ClusterVizService.get_cluster_group(params, @study)
```